frontend/catalyst/utils/toml.py

```diff
@@ -52,25 +55,65 @@ def read_toml_file(toml_file: str) -> TOMLDocument:
     return config
 
 
-def check_mid_circuit_measurement_flag(config: TOMLDocument) -> bool:
-    """Check the global mid-circuit measurement flag"""
-    return bool(config.get("compilation", {}).get("mid_circuit_measurement", False))
+@dataclass(unsafe_hash=True)
+class OperationProperties:
+    """Capabilities of a single operation"""
 
+    invertible: bool
+    controllable: bool
+    differentiable: bool
 
-def check_adjoint_flag(config: TOMLDocument, shots_present: bool) -> bool:
-    """Check the global adjoint flag for toml schema 1. For newer schemas the adjoint flag is
-    defined to be set if all native gates are inverible"""
-    schema = int(config["schema"])
-    if schema == 1:
-        return bool(config.get("compilation", {}).get("quantum_adjoint", False))
 
-    elif schema == 2:
-        return all(
-            "invertible" in v.get("properties", {})
-            for g, v in get_native_gates(config, shots_present).items()
-        )
+def intersect_properties(a: OperationProperties, b: OperationProperties) -> OperationProperties:
+    """Calculate the intersection of OperationProperties"""
+    return OperationProperties(
+        invertible=a.invertible and b.invertible,
+        controllable=a.controllable and b.controllable,
+        differentiable=a.differentiable and b.differentiable,
+    )
+
+
+@dataclass
+class DeviceConfig:
+    """Quantum device capabilities"""
+
+    native_gates: Dict[str, OperationProperties]
```

Comment:
We do not use the term gates too much in PL. What about: `native_ops`, `to_decompose_ops`, `to_matrix_ops` and `obs`? @dime10

Reply:
I am neutral to names, just let me know the right ones.

Reply:
Renamed